pycalphad/core/eqsolver.pyx

```diff
@@ -235,6 +278,8 @@ def _solve_eq_at_conditions(properties, phase_records, grid, conds_keys, state_v
             composition_sets.append(compset)
         chemical_potentials = prop_MU_values[it.multi_index]
         energy = prop_GM_values[it.multi_index]
+        add_nearly_stable(composition_sets, phase_records, grid, curr_idx, chemical_potentials,
```

Comment:
Does it make sense that the nearly stable composition sets have one shot to make it into the solution? After the first call to `find_solution`, all the remaining metastable composition sets will be removed if `find_solution` converged and the nearly metastable composition sets won't be able to make it in.

Since we have more metastable composition sets in the solution, does it make sense to remove the metastable composition sets less aggressively after `find_solution`, waiting until we think we found the global min to remove the metastable candidates?

Reply:
Following up on the previous comment, can we replace `add_new_phases` by `add_nearly_stable` if we slightly modify the behavior of `add_nearly_stable` to add a small amount of a phase with positive driving force (like the 1e-6 we use in `add_new_phases`)?

Reply:
Including calls to `add_nearly_stable` inside the main solve loop seems to cause convergence issues for `test_eq_stepsize_reduction`. Setting `remove_metastable=False` inside the loop and putting a separate call to `update_composition_sets` at the end also causes a convergence failure in that test. I haven't fully run the issue down. Because the logic for how phases are changed is only a heuristic, it's possible to fail to come up with the stable set of phases in a finite number of iterations.